js/src/Components/Widgets/LinkWidget.js

```diff
@@ -16,18 +16,22 @@ export const LinkWidget = ({fieldId, defaultValue, onFieldChange, settings}) =>
     title: defaultValue && defaultValue.length ? defaultValue[0].title : ''
   };
 
+  /**
+   * When the uri changes, use a timer like a debounce and fetch some suggestions from the linkit module.
+   */
   const uriChanged = (newUri) => {
     clearTimeout(timeout);
 
     // Make a new timeout set to go off in 800ms
     timeout = setTimeout(() => {
 
-      onFieldChange([{
-        title: defaultFieldValue.title,
-        uri: getUriFromString(newUri)
-      }]);
-
-      if (newUri.substr(0, 1) === '/' || newUri.substr(0, 1) === '<') {
+      // If the user enters a url that starts with some characters, we dont want to fetch suggestions that will just
+      // be empty anyways. This includes absolute urls, <front> and relative urls.
+      if (
+        newUri.substr(0, 1) === '/' ||
+        newUri.substr(0, 1) === '<' ||
+        newUri.substr(0, 4) === 'http'
```

Comment:
What about other protocols? ftp, smb, or streaming URLs? Perhaps validate for the pattern `[a-z]*:\/\/`

Reply:
do people put those in there?!

Reply:
Someone somewhere has a use case for it and it wouldn't be too much for us to allow it here.

Reply:
well its not that it "doesnt allow" this just avoids unnecessary suggestion api calls. but i get what you're saying. no need to provide suggestions for ftp urls

Reply:
I swapped with that protocol regex check